### PR TITLE
Add `Integer.Format.Unicode`

### PR DESCRIPTION
**Pull request by mneumann**
Related issue: #350

### CODE REVIEW

File: spec/core/Integer.Format.Spec.savi [hunk 1]
```diff
@@ -49,3 +49,7 @@
     assert: "\(U8[0].format.bin.without_leading_zeros)" == "0b0"
     assert: "\(I16[36].format.bin.without_leading_zeros)" == "0b100100"
     assert: "\(I16[-36].format.bin.without_leading_zeros)" == "0b1111111111011100"
+
+  :it "can format integers as Unicode codepoints"
+    assert: "\('S'.format.unicode)" == "S"
+    assert: "\(U8[83].format.unicode)" == "S"
```
Comment (jemc) [inline spec/core/Integer.Format.Spec.savi:55]: Can you also add at least one test for a non-ASCII unicode character?

File: core/Integer.Format.savi [hunk 2]
```diff
@@ -8,6 +8,8 @@
   :fun hexadecimal: Integer.Format.Hexadecimal(T)._new(@_value)
   :fun binary: Integer.Format.Binary(T)._new(@_value)
```
Comment (jemc) [inline core/Integer.Format.savi:10]: I'm just suggesting here to remove this empty line above, so that it gets grouped with the similar methods above.
```suggestion
```

File: spec/core/Integer.Format.Spec.savi (hunk 1 above)
Comment (mneumann) [reply spec/core/Integer.Format.Spec.savi:55]: can I use `\U` escape sequences in Savi?

Comment (jemc) [reply spec/core/Integer.Format.Spec.savi:55]: Yes, but only with a lowercase `\u` - followed by 4 hexadecimal digits

**Review by jemc — APPROVED** [review]
Looks good. Either one of us can merge this after CI passes (ignoring the FreeBSD build that is failing today).

**Comment by jemc** [comment]
@mneumann - For some reason, your PR didn't run the two GitHub Actions CI checks that actually run the test suite - only the Cirrus CI per-platform build tests ran...

So we actually merged this with a bug. See https://github.com/savi-lang/savi/pull/355 for details.

On your next PR I want to figure out how to get the two GitHub Actions checks to run properly.
